test/Dapper.AOT.Test/Interceptors/CommandDefinition.output.cs

```diff
@@ -0,0 +1,193 @@
+            // returns data: string
+            global::System.Diagnostics.Debug.Assert(!string.IsNullOrWhiteSpace(command.CommandText));
+            global::System.Diagnostics.Debug.Assert((command.CommandType ?? global::Dapper.DapperAotExtensions.GetCommandType(command.CommandText)) == global::System.Data.CommandType.Text);
+            global::System.Diagnostics.Debug.Assert(command.Buffered is false);
+            global::System.Diagnostics.Debug.Assert(command.Parameters is not null);
+
+            return global::Dapper.DapperAotExtensions.Command(cnn, command.Transaction, command.CommandText, command.CommandType ?? default, command.CommandTimeout ?? default, CommandFactory0.Instance).ExecuteScalarAsync<string>(command.Parameters, cancellationToken: command.CancellationToken);
+
+        }
+
+        [global::System.Runtime.CompilerServices.InterceptsLocationAttribute("Interceptors\\CommandDefinition.input.cs", 38, 24)]
+        internal static string? ExecuteScalar3(this global::System.Data.IDbConnection cnn, global::Dapper.CommandDefinition command)
+        {
+            // Execute, TypedResult, HasParameters, Text, Scalar, KnownParameters
+            // takes parameter: <anonymous type: int X>
+            // parameter map: X
+            // returns data: string
+            global::System.Diagnostics.Debug.Assert(!string.IsNullOrWhiteSpace(command.CommandText));
+            global::System.Diagnostics.Debug.Assert((command.CommandType ?? global::Dapper.DapperAotExtensions.GetCommandType(command.CommandText)) == global::System.Data.CommandType.Text);
+            global::System.Diagnostics.Debug.Assert(command.Buffered is false);
+            global::System.Diagnostics.Debug.Assert(command.Parameters is not null);
+
+            return global::Dapper.DapperAotExtensions.Command(cnn, command.Transaction, command.CommandText, command.CommandType ?? default, command.CommandTimeout ?? default, CommandFactory0.Instance).ExecuteScalar<string>(command.Parameters);
+
+        }
+
+        [global::System.Runtime.CompilerServices.InterceptsLocationAttribute("Interceptors\\CommandDefinition.input.cs", 48, 24)]
+        internal static global::System.Threading.Tasks.Task<string?> ExecuteScalarAsync4(this global::System.Data.IDbConnection cnn, global::Dapper.CommandDefinition command)
+        {
+            // Execute, Async, TypedResult, Scalar
+            // returns data: string
+            global::System.Diagnostics.Debug.Assert(!string.IsNullOrWhiteSpace(command.CommandText));
+            global::System.Diagnostics.Debug.Assert(command.Buffered is false);
+            global::System.Diagnostics.Debug.Assert(command.Parameters is null);
+
+            return global::Dapper.DapperAotExtensions.Command(cnn, command.Transaction, command.CommandText, command.CommandType ?? default, command.CommandTimeout ?? default, DefaultCommandFactory).ExecuteScalarAsync<string>(command.Parameters, cancellationToken: command.CancellationToken);
```

Comment:
1) check if it works at all
2) if it does not and we cant detect it; we should let user know via analyzer warning / just not generate AOT out of it
3) if it does - its ok

Reply:
if `parameters` is null - then it is probably fine, we will not fallback to any specific type handler